posts/2019-07-04-Rust-1.36.0.md

```diff
@@ -0,0 +1,164 @@
+
+With NLL on both editions, we are closer to removing the old borrow checker. However, the old borrow checker unfortunately accepted some [unsound][soundness] code it should not have. As a result, NLL is currently in a "migration mode" wherein we will emit warnings instead of errors if the NLL borrow checker rejects code the old AST borrow checker would accept. Please see [this list][crater-nll] to see if one of your crates are affected.
+
+To read more about NLL, MIR, the story around fixing soundness holes, and what you can do about the warnings if you have them, [read Felix Klock's blog post][felix-blog].
+
+### A new `HashMap<K, V>` implementation
+
+[`hashbrown`]: https://crates.io/crates/hashbrown
+[`HashMap<K, V>`]: https://doc.rust-lang.org/std/collections/struct.HashMap.html
+[pr-hashbrown]: https://github.com/rust-lang/rust/pull/58623
+[SwissTable]: https://abseil.io/blog/20180927-swisstables
+[pr-hashbrown-perf]: https://perf.rust-lang.org/compare.html?start=b57fe74a27590289fd657614b8ad1f3eac8a7ad2&end=abade53a649583e40ed07c26ee10652703f09b58&stat=wall-time
+
+In Rust 1.36.0, the `HashMap<K, V>` implementation has been [replaced][pr-hashbrown] with the one in the [`hashbrown`] crate which is based on the [SwissTable] design. While the interface is the same, the `HashMap<K, V>` implementation is now on [average faster][pr-hashbrown-perf] and has lower memory overhead. Note that unlike the `hashbrown` crate, the implementation in `std` continues to use SipHash 1-3.
+
+### [`--offline`] support in Cargo
+
+[`--offline`]: https://doc.rust-lang.org/cargo/commands/cargo-build.html#cargo_build_manifest_options
+[`cargo fetch`]: https://doc.rust-lang.org/cargo/commands/cargo-fetch.html
+
+During most builds, Cargo doesn't interact with the network. Sometimes, however, Cargo has to. Such is the case when a dependency is added and the latest compatible version needs to be downloaded. At times, network access is not an option though, for example on an airplane or in isolated build environments.
+
+In Rust 1.36, a new Cargo flag has been stabilized: [`--offline`]. The flag alters Cargo's dependency resolution algorithm to only use locally cached dependencies. When the required ones are not available offline, and a network access would be required, Cargo will error out. To prepopulate the local cache, it's possible to use the [`cargo fetch`] command, which downloads all the required dependencies beforehand.
+
+[relnotes-cargo]: https://github.com/rust-lang/cargo/blob/master/CHANGELOG.md#cargo-136-2019-07-04
+
+For more information on other changes to Cargo, see the [detailed release notes][relnotes-cargo].
+
+### Library changes
+
+[`dbg!`]: https://doc.rust-lang.org/std/macro.dbg.html
+
+The [`dbg!`] macro now supports multiple arguments.
+
+Moreover, a number of APIs have been made `const`:
```

Comment:
```suggestion
A number of APIs have been made `const`:
```

Reply:
The reason for having "Moreover, ..." here is that I find that it doesn't flow well as prose otherwise if you just start with "A..."

Reply:
That's fair enough, but it's not consistent with lines 133 and 153. If we do have a transition word, I'd suggest "additionally", because I usually think of "moreover" as implying a closer relationship between the two sentences than what we have here.

Reply:
Alright; let's go with "additionally" then. :)